pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -90,6 +90,54 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 			{
 				Name: "kubevirt.rules",
 				Rules: []v1.Rule{
+					{
+						Record: "virt_controller_rest_client_requests",
```

Comment:
curious, do we really need these additional metrics? If we are not going to use it anywhere else - may be it would be better to just update the logic for existing metrics

Reply:
Hey,
The motivations of defining these functions and records:
1)  `GeneratePromQlQueryForValueDifferenceInFiveMinutesInterval` is much more declarative and clearer than 
`clamp_min(sum(rest_client_requests_total{pod=~'virt-controller-.*'}) - sum(rest_client_requests_total{pod=~'virt-controller-.*')} offset 5m), 0)`.
2) if we will want to change the way we calculate the difference in the future with function we would have to modify only one line whereas without it we would have to modify 8 lines.
3) without functions we increase the chance to have typo in the future because of the IDE's code auto completion and the code won't compile if we would have a typo in the function name In contrast to string.

Reply:
I agree that using a function is helpful here, but I also wonder if we need extra metrics.

What I would suggest is create another function instead of defining new metrics. For example, you can introduce a `generateAllRestRequestsQuery(pod, namespace string) string` function. This function, as its name suggests, can generate a query without adding another `Record` metric.

For example, calling `generateAllRestRequestsQuery("virt-operator", "my-namespace")` could return `"rest_client_requests_total{pod=~'virt-operator-.*', namespace=my-namespace}"`. Then, as @dshchedr said, you can only edit existing records by doing something like: `GeneratePromQlQueryForValueDifferenceInFiveMinutesInterval(generateAllRestRequestsQuery("virt-operator", "my-namespace"))`.

Reply:
of course, another function can be made for rest calls with errors.

Reply:
yes your right it might be better to use variable or another function instead of records. 
Thanks!